ERCS/erc-7574.md

```diff
@@ -0,0 +1,191 @@
+
+
+## Specification
+
+<!--
+  The Specification section should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (besu, erigon, ethereumjs, go-ethereum, nethermind, or others).
+
+  It is recommended to follow RFC 2119 and RFC 8170. Do not remove the key word definitions if RFC 2119 and RFC 8170 are followed.
+
+  TODO: Remove this comment before submitting
+-->
+
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+
+### MetaData Interface
+
+
+NFT metadata structured by the official [ERC-721](./eip-721.md) metadata standard or the Enjin Metadata suggestions are as follows referring to OpenSea docs.
+
+```jsx
+{
+   "description":"University bachelor KYC",
+   "external_url":"https://api.kor.credential/metadata/1/1",
+   "home_url":"https://app.kor.credential/token/1",
+   "image_url":"https://www.poap.xyz/events/badges/ethdenver-19.png",
+   "name":" KOREA KYC",
+   "attributes" : { ... },
+}
+```
+
+In addition to the existing NFT metadata standards, a new standard for identity verification supplements the metadata with the "Issuer" and "credentialNumber" sections. When issuing SBTs, the Issuer and the credential number from the Credential are specified. This approach ensures that the user's personal information remains private. In case of any issues, the identity verification process can be traced back by requesting the credential number used for verification from the Issuer without revealing the user's information. This standard represents an extension of ERC-721 metadata.
+
+```jsx
+{
+   "description":"University bachelor KYC",
+   "external_url":"https://api.kor.credential/metadata/1/1",
+   "home_url":"https://app.kor.credential/token/1",
+   "image_url":"https://www.poap.xyz/events/badges/ethdenver-19.png",
+   "name":" KOREA KYC",
+   "attributes" : { ... },
+   "Iusser" : { ...},
+   "credentialNumber" : { ... },
+ }
+```
+
+We have defined a structure to represent VerifiedPresentation. It includes the user's wallet address, issuer, user name, and Credential Number
+
+```jsx
+struct VerifiedPresentation {
+address userAddr;
+string issuer;
+string user;
+uint CredentialNumber;
+}
+```
+
+Following that, here is the contract that verifies a user's Credential proof and issues SBTs:
+
+
+### Contract Interface
+
+
+In this scenario, a **`verify`** function is created using Zokrates. Upon successful verification, the generated **`verify`** function calls the SBT issuance contract, enabling the entire process.
```

Comment:
> using Zokrates

Hm, so this definitely doesn't provide enough information to implementers. If the format of the `Proof` structure is "whatever Zokrates generates", then everyone implementing this standard has to use Zokrates, and there isn't much point of having an ERC at all.

ERCs really shine when multiple independent parties have to coordinate without otherwise communicating. ERC-20 is a great example. Every token wants to work with every wallet, and vice versa; perfect for an ERC.